hpx/parallel/util/numa_allocator.hpp

```diff
@@ -124,6 +124,11 @@ namespace hpx { namespace parallel { namespace util
             }
             hpx::wait_all(first_touch);
 
+            for (auto && f : first_touch)
+            {
+                f.get();        // rethrow exceptions
+            }
+
```

Comment:
I think this officially exposes the brokenness of our NUMA handling(?) on CircleCI (https://circleci.com/gh/STEllAR-GROUP/hpx/86317). #3802 also had problems with this. We might just have to disable or remove this example and leave it to @biddisco to replace this `numa_allocator` with his new and improved one ;)